beesdoo_product/models/beesdoo_product.py

```diff
@@ -272,18 +299,28 @@ def _unit_same_category(self):
                     )
                 )
 
+    # fixme rename to _compute_suggested_price
+    # fixme move to new module product_suggested_price
+    #  or sale_suggested_price
     @api.multi
-    @api.depends("seller_ids", "supplier_taxes_id", "taxes_id")
+    @api.depends(
+        "seller_ids",
+        "supplier_taxes_id",
+        "taxes_id",
+        "uom_id",
+        "uom_po_id",
+        "categ_id.should_round_suggested_price",
+    )
     def _compute_cost(self):
         suggested_price_reference = (
             self.env["ir.config_parameter"]
             .sudo()
             .get_param("beesdoo_product.suggested_price_reference")
         )
         for product in self:
-            suppliers = product._get_main_supplier_info()
-            if len(suppliers) > 0:
-                price = suppliers[0].price
+            supplier = product._get_main_supplier_info()
```

Comment:
could `product.main_seller_id` be used instead? it seems that “seller” and “supplier” are used interchangeably in this class. is this on purpose?

looking at `_get_main_supplier_info()`, it took a little time to understand the logic. a “seller” with no `.date_start` comes before the others? tip in that function: instead of defining `far_future` with a magic date, use `date.max` directly. also, this function either returns a supplier or a collection. wouldn’t it be more logical to return a supplier or `None`?

Reply:
> could product.main_seller_id be used instead? it seems that “seller” and “supplier” are used interchangeably in this class. is this on purpose?

Indeed, it would need a little cleanup. I left a comment to deal with it in the upcoming refactor done for PYSI.

> looking at _get_main_supplier_info(), it took a little time to understand the logic. a “seller” with no .date_start comes before the others?

Yes, I guess that comes from a specification that would have gained to be in a code comment. I added a comment.

> tip in that function: instead of defining far_future with a magic date, use date.max directly.

TIL 👍 I fixed it

> also, this function either returns a supplier or a collection. wouldn’t it be more logical to return a supplier or None?

Not sure about that. Another way to see it : the functions always returns a RecordSet : a RecordSet of 1 `supplierinfo` or an empty RecordSet. I left a comment in the code anyway.

I won't fix further since this is an old task that has been tested by Virginie and Pol. But I left comment that could be delt w/ in PYSI.